radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/addressbook/AddressBookEntry.java

```diff
@@ -288,33 +293,33 @@ public int hashCode() {
 		@DsonOutput(DsonOutput.Output.ALL)
 		private final RadixNodeUri uri;
 
-		private final Optional<Instant> lastSuccessfulConnection;
+		private final Optional<LatestConnectionStatus> latestConnectionStatus;
 
 		private final Optional<Instant> blacklistedUntil;
 
 		@JsonCreator
 		private static PeerAddressEntry deserialize(
 			@JsonProperty("uri") RadixNodeUri uri,
-			@JsonProperty("lastSuccessfulConnection") Long rawLastSuccessfulConnection,
+			@JsonProperty("latestConnectionStatus") String rawLatestConnectionStatus,
 			@JsonProperty("blacklistedUntil") Long rawBlacklistedUntil
 		) {
-			final var lastSuccessfulConnection = Optional.ofNullable(rawLastSuccessfulConnection).map(Instant::ofEpochMilli);
+			final var latestConnectionStatus = Optional.ofNullable(rawLatestConnectionStatus).map(LatestConnectionStatus::valueOf);
```

Comment:
what if `valueOf` fails?

Reply:
it'll fail with a runtime exception, do you think we should handle it differently?
this class is not deserialized from an

Reply:
Not sure we can assume the string will always be correct. if it is an optional field, we could log it as a warn and fallback to Optinal.empty() instead of throwing an exception. What do you think?

Reply:
I'm not sure, I think I'd rather fail with an error cause this would mean there's either a database corruption or a serious serialization bug

Reply:
makes sense.